vaadin-icons-flow-demo/src/main/resources/META-INF/resources/frontend/all-icons.html

```diff
@@ -0,0 +1,157 @@
+<!-- Added Web Component dependencies to make Vaadin Designer preview work -->
+
+<!-- Defines the reviews-list element -->
+<dom-module id="all-icons">
+  <template>
+    <style>
+      :host {
+        display: block;
+      }
+
+      dom-repeat {
+        display: block;
+        width: 100%;
+      }
+
+      .iconset-container {
+        margin-top: 1em;
+        background-Color: #fff;
+      }
+
+      .icon-container {
+        margin: 18px 0;
+        padding: 0 5px;
+        text-align: center;
+        width: 19.3%;
+        display: inline-block;
+        cursor: default;
+        vertical-align: top;
+        box-sizing: border-box;
+      }
+
+      .wrapper-icon-label {
+          display: flex;
+          flex-direction: column;
+          align-items: center;
+      }
+
+      .icon {
+        width: 50px !important;
+        height: 50px !important;
+        margin-bottom: 3px;
+        fill: #00b4f0 !important
+      }
+
+      .text-icon {
+        margin: 5px;
+        width: 140px;
+        font-size: 12px;
+      }
+
+      @media (max-width: 1024px) {
+        .icon-container {
+          width: 24%;
+        }
+      }
+
+      @media (max-width: 560px) {
+        .icon-container {
+          width: 32%;
+        }
+      }
+
+      @media (max-width: 420px) {
+        .icon-container {
+          width: 49%;
+        }
+      }
+
+      .icon-container span {
+        font-family: sans-serif;
+        font-size: 12px;
+        display: block;
+        margin-top: .5em;
+        color: #616161;
+        transition: all 600ms;
+      }
+
+      .icon-container:hover span {
+        color: #00b4f0;
+      }
+
+      .icon-container iron-icon {
+        display: inline-block;
+        fill: #616161;
+        transition: all 200ms;
+        width: 30px;
+        height: 30px;
+      }
+
+      .icon-container:hover iron-icon {
+        fill: #00b4f0;
+      }
+    </style>
+    <div class="iconset-container">
+     <dom-repeat items="[[icons]]">
+        <template>
+          <div class="icon-container">
+            <div class="wrapper-icon-label">
+                <iron-icon icon="[[item]]" class="icon"></iron-icon>
+                <span class="text-icon">[[getNameIcons(item)]]</span>
+            </div>
+          </div>
+        </template>
+      </dom-repeat>
+    </div>
+  </template>
+
+  <!-- Polymer boilerplate to register the reviews-list element -->
+  <script>
+    class AllIconsView extends Polymer.Element {
+      static get is() {
+        return 'all-icons';
+      }
+
+      static get properties() {
+        return {
+          icons : Array
+        }
+      }
+
+      ready() {
+        super.ready();
+        const iconset = new Polymer.IronMeta({type: 'iconset'}).list.filter(i => i.name == 'vaadin')[0];
+        this.icons = iconset.getIconNames();
+      }
+
+      getNameIcons(iconset) {
+          return iconset.split(':')[1].toString().toUpperCase().split('-').join('_');
```

Comment:
Doing 2 consecutive `split` for more than 600+ icon names is probably not the best idea.
This is sort of a nitpick, but considering the that the reason for changing the demo is to improve performance maybe we should consider it.

Reply:
I changed it. Using `getNameIcons(iconset) {
          const labelIcon = iconset.substring(iconset.indexOf(":") + 1);
          return labelIcon.toUpperCase().replace(/-/g, '_');
      }` instead of using split 2 times.

I think the performance is improved. Because using split, we have 3 steps: find the replacement, split to 2 arrays and join it. But with replace, we find the replacement and replace immediately.

Please correct me if I am wrong. Thank you.